src/py/mat3ra/made/tools/build/utils.py

```diff
@@ -0,0 +1,87 @@
+from typing import List, Dict
+
+import numpy as np
+from mat3ra.made.material import Material
+from mat3ra.made.utils import filter_array_with_id_value_by_ids
+
+from mat3ra.made.tools.utils import convert_basis_to_crystal
+
+
+def resolve_close_coordinates(basis: Dict, distance_tolerance: float = 0.01) -> Dict:
+    """
+    Resolve close coordinates in the basis by replacing them with the latest one.
+    """
+    elements = basis["elements"]
+    coordinates = basis["coordinates"]
+    labels = basis.get("labels", [])
+
+    ids = set([coord["id"] for coord in coordinates])
+    ids_to_remove = set()
+
+    for i in range(1, len(coordinates)):
```

Comment:
We should use pymatgen or ase method here

Reply:
Haven't seen anything that is exactly what we need here. This deals with ids in array, that neither pmg or ase do